File: app/src/main/java/com/trayis/simpliretrodemo/model/ReposData.java
package com.trayis.simpliretrodemo.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by mudesai on 9/18/16.
 */
public class ReposData implements Parcelable {

    public static final Creator<ReposData> CREATOR = new Creator<ReposData>() {
        @Override
        public ReposData createFromParcel(Parcel in) {
            return new ReposData(in);
        }

        @Override
        public ReposData[] newArray(int size) {
            return new ReposData[size];
        }
    };
    public Repo[] repos;

    protected ReposData(Parcel in) {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }
}

File: app/src/main/java/com/trayis/simpliretrodemo/model/ReposData.kt
package com.trayis.simpliretrodemo.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by mudesai on 9/18/16.
 */
data class ReposData protected constructor(val repos: Array<Repo>? = null) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.createTypedArray(Repo)) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeTypedArray(repos, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ReposData

        if (repos != null) {
            if (other.repos == null) return false
            if (!repos.contentEquals(other.repos)) return false
        } else if (other.repos != null) return false

        return true
    }

    override fun hashCode(): Int {
        return repos?.contentHashCode() ?: 0
    }

    companion object CREATOR : Parcelable.Creator<ReposData> {
        override fun createFromParcel(parcel: Parcel): ReposData {
            return ReposData(parcel)
        }

        override fun newArray(size: Int): Array<ReposData?> {
            return arrayOfNulls(size)
        }
    }

}
